transformer_engine/pytorch/csrc/userbuffers/userbuffers.h

```diff
@@ -54,6 +61,23 @@
 #define NVTE_HF_NVREDUCEDONE (userbuffers_op_types + 3)
 #define NVTE_MAX_SHARP 16
 
+// Strip the path from a full filename
+#define FILENAME(file) ({ \
+    const char* filename = file; \
+    const char* basename = filename; \
+    for (const char* ptr = filename; *ptr != '\0'; ptr++) { \
+        if (*ptr == '/' || *ptr == '\\') { \
+            basename = ptr + 1; \
+        } \
+    } \
+    basename; \
+})
+
+// Printf to provide enough information so it is easier to attribute failures
+#define UB_PRINT(message, ...) printf("[%s:%s:%d] " message "\n", FILENAME(__FILE__),              \
+                                                                  __FUNCTION__,                    \
+                                                                  __LINE__, __VA_ARGS__)
```

Comment:
This header is imported outside of UB, so a macro name like `FILENAME` could experience a name conflict. Also, printing the full path seems fine for error messages:
```suggestion
// Printf to provide enough information so it is easier to attribute failures
#define UB_PRINT(message, ...) printf("[%s:%s:%d] " message "\n", __FILE__,                        \
                                                                  __FUNCTION__,                    \
                                                                  __LINE__, __VA_ARGS__)
```

Reply:
We had a version printing out full path and it was polluting log file. Since we have uniquely names files for ub there is no need for full path. It makes sense to rename the macro.